src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -340,11 +350,24 @@ export class EditDataGridPanel extends GridParentComponent {
 
 		// At the end of a successful cell select, update the currently selected cell
 		cellSelectTasks = cellSelectTasks.then(() => {
+			this.disableEnableSelect(true);
 			self.setCurrentCell(row, column);
+			self.focusCell(row, column);
 		});
 
 		// Cap off any failed promises, since they'll be handled
-		cellSelectTasks.catch(() => { });
+		cellSelectTasks.catch(() => {
+		});
+	}
+
+	// Disables editing the grid temporarily when clicking on a cell (to allow for any processing tasks to be finished first such as adding a new row).
+	private disableEnableSelect(state: boolean) {
```

Comment:
1. please this.table.grid.setOptions and change the editable option. 
2. rename this method to updateEnabledState(state:boolean)

Reply:
The table.grid.setOptions triggered an infinite loop due to calling render every time it is called.